pkg/frontend/querymiddleware/astmapper/sharding_test.go

```diff
@@ -496,46 +512,36 @@ func TestShardSummer(t *testing.T) {
 		{
 			in:                     `sum(rate(metric[1m])) and max(metric) > 0`,
 			out:                    `sum(` + concatShards(t, 3, `sum(rate(metric{__query_shard__="x_of_y"}[1m]))`) + `) and max(` + concatShards(t, 3, `max(metric{__query_shard__="x_of_y"})`) + `) > 0`,
-			expectedShardedQueries: 6,
+			expectedShardedQueries: 2,
 		},
 		{
 			in:                     `sum(rate(metric[1m])) > avg(rate(metric[1m]))`,
 			out:                    `sum(` + concatShards(t, 3, `sum(rate(metric{__query_shard__="x_of_y"}[1m]))`) + `) > (sum(` + concatShards(t, 3, `sum(rate(metric{__query_shard__="x_of_y"}[1m]))`) + `) / sum(` + concatShards(t, 3, `count(rate(metric{__query_shard__="x_of_y"}[1m]))`) + `))`,
-			expectedShardedQueries: 9,
+			expectedShardedQueries: 3,
 		},
 		{
 			in:                     `group by (a, b) (metric)`,
 			out:                    `group by (a, b) (` + concatShards(t, 3, `group by (a, b) (metric{__query_shard__="x_of_y"})`) + `)`,
-			expectedShardedQueries: 3,
+			expectedShardedQueries: 1,
 		},
 		{
 			in:                     `count by (a) (group by (a, b) (metric))`,
 			out:                    `count by (a) (group by (a, b) (` + concatShards(t, 3, `group by (a, b) (metric{__query_shard__="x_of_y"})`) + `))`,
-			expectedShardedQueries: 3,
+			expectedShardedQueries: 1,
```

Comment:
Given how this is being used maybe it makes more sense to call it `expectedShardableQueries` since it's now the number of queries that could be sharded in the original query (instead of the total number of queries including all their shards)?